homeassistant/components/fibaro/__init__.py

```diff
@@ -65,30 +66,38 @@
 
 FIBARO_ID_LIST_SCHEMA = vol.Schema([cv.string])
 
+GATEWAY_CONFIG = vol.Schema({
+    vol.Required(CONF_PASSWORD): cv.string,
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_URL): cv.url,
+    vol.Optional(CONF_PLUGINS, default=False): cv.boolean,
+    vol.Optional(CONF_EXCLUDE, default=[]): FIBARO_ID_LIST_SCHEMA,
+    vol.Optional(CONF_DEVICE_CONFIG, default={}):
+        vol.Schema({cv.string: DEVICE_CONFIG_SCHEMA_ENTRY})
+}, extra=vol.ALLOW_EXTRA)
+
 CONFIG_SCHEMA = vol.Schema({
     DOMAIN: vol.Schema({
-        vol.Required(CONF_PASSWORD): cv.string,
-        vol.Required(CONF_USERNAME): cv.string,
-        vol.Required(CONF_URL): cv.url,
-        vol.Optional(CONF_PLUGINS, default=False): cv.boolean,
-        vol.Optional(CONF_EXCLUDE, default=[]): FIBARO_ID_LIST_SCHEMA,
-        vol.Optional(CONF_DEVICE_CONFIG, default={}):
-            vol.Schema({cv.string: DEVICE_CONFIG_SCHEMA_ENTRY})
+        vol.Optional(CONF_GATEWAYS, default=[{}]):
+            vol.All(cv.ensure_list, [GATEWAY_CONFIG])
     })
 }, extra=vol.ALLOW_EXTRA)
 
 
 class FibaroController():
     """Initiate Fibaro Controller Class."""
 
-    def __init__(self, username, password, url, import_plugins, config):
+    def __init__(self, config):
         """Initialize the Fibaro controller."""
         from fiblary3.client.v4.client import Client as FibaroClient
-        self._client = FibaroClient(url, username, password)
+
+        self._client = FibaroClient(config.get(CONF_URL),
```

Comment:
Don't use `dict.get` for required config keys or keys with default config schema values. Use `dict[key]`.

Reply:
Thanks, fixed